workers/unity/Packages/io.improbable.gdk.buildsystem/WorkerBuilder.cs

```diff
@@ -68,8 +69,20 @@ internal static void MenuBuild(BuildEnvironment environment, params string[] wor
             {
                 try
                 {
-                    var buildContexts = BuildContext.GetBuildContexts(workerTypes, environment);
-                    BuildWorkers(buildContexts);
+                    var overrideCompilerError = CompilerValidator.HasCompileErrors
+                        && EditorUtility.DisplayDialog("Compile Errors in Scripts",
+                            "There are outstanding compile errors, are you sure you want to build the workers?",
+                            "Yes", "No");
+
+                    if (!CompilerValidator.HasCompileErrors || overrideCompilerError)
+                    {
+                        var buildContexts = BuildContext.GetBuildContexts(workerTypes, environment);
+                        BuildWorkers(buildContexts);
+                    }
+                    else
+                    {
+                        throw new BuildFailedException($"Compiling failed with {CompilerValidator.CompileErrorCount} errors.");
+                    }
```

Comment:
I'd possibly invert this if-else block: 

```csharp
if (CompilerValidtor.HasCompileErrors && !overrideCompilerError)
{
	throw new BuildFailedException($"Cannot build worker as compile failed with {CompilerValidator.CompileErrorCount} errors.");
}

var buildContexts = BuildContext.GetBuildContexts(workerTypes, environment);
BuildWorkers(buildContexts);
```

There's also an argument that rather than throwing an exception, it should just return as the user has already been alerted. It may be a bit odd to get the first dialog, hit "No", then get a second dialog (in the `catch` block) telling you the build failed.

Reply:
Yeah, I would agree that it is a nicer UX to cancel the build when they Click "No" instead of seeing the standard "Build Failed" dialog